includes/class-wc-stripe-helper.php

```diff
@@ -2044,4 +2044,22 @@ public static function is_connected( $mode = null ) {
 	public static function is_stripe_gateway_order( $order ) {
 		return WC_Stripe_UPE_Payment_Gateway::ID === substr( (string) $order->get_payment_method(), 0, 6 );
 	}
+
+	/**
+	 * Checks if the enhanced debug mode is enabled.
+	 *
+	 * @return bool True if enabled, false otherwise.
+	 */
+	public static function is_enhanced_debug_mode_enabled(): bool {
+		/**
+		 * Filters the flag that decides if the enhanced debug mode is enabled.
+		 *
+		 * @since 10.1.0
+		 *
+		 * @param bool $enabled True if enabled, false otherwise.
+		 *
+		 * @return bool True if enabled, false otherwise.
+		*/
+		return apply_filters( 'wc_stripe_enhanced_debug_mode_enabled', false );
+	}
```

Comment:
From a naming perspective, I think it would be better if we use a more specific name relating to OAuth or connections, as "enhanced debug mode" is pretty generic, and I would think that we'd generally want to continue using specific flags in future cases where we want to enable detailed debugging related to a specific feature or area of the code. So I would vote for `is_connection_debug_mode_enabled()` or `is_oauth_debug_mode_enabled()` for the method name, and then carry that through to the filter name.